components/announcement_bar/renewal_link/renewal_link.tsx

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+
+import {Client4} from 'mattermost-redux/client';
+import {FormattedMessage} from 'react-intl';
+
+import {trackEvent} from 'actions/telemetry_actions';
+import {
+    ModalIdentifiers,
+} from 'utils/constants';
+
+import NoInternetConnection from '../no_internet_connection/no_internet_connection';
+
+export interface RenewalLinkProps {
+    actions: {
+        openModal: (modalData: { modalId: string; dialogType: any; dialogProps?: any }) => void;
+    };
+}
+
+const RenewalLink: React.FC<RenewalLinkProps> = (props: RenewalLinkProps) => {
+    const [renewalLink, setRenewalLink] = useState('https://mattermost.com/renew/');
+
+    useEffect(() => {
+        Client4.getRenewalLink().then(({renewal_link: renewalLinkParam}) => {
+            if (renewalLinkParam && (/^http[s]?:\/\/customers\.mattermost\.com\/subscribe\/renew/).test(renewalLinkParam)) {
+                setRenewalLink(renewalLinkParam);
+            }
+        });
+    }, []);
+
+    const handleLinkClick = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+        e.preventDefault();
+        trackEvent('announcement_bar', 'renew_license');
```

Comment:
For the telemetry we usually group all related events in the same category, first param, and then the event name would be the action that was performed at the time, so based on https://docs.google.com/document/d/1HgZpqadMKvvSZdWoD_blysRwjdgqFbqn2o4zJNZGhYc/edit# I would suggest: 
```suggestion
        trackEvent('renew_license', 'renew_license_banner_success');
```